### ISSUE TITLE
class labels need a format?

### ISSUE THREAD

**Issue by stevcabello** [crflynn/skranger#44]
Hi, 

Do class labels need to have a specific format? For instance, should they start with 0,1,2....... or something else?

I was comparing RangerForestClassifier against scikit-learn's RandomForestClassifier by using some time series datasets from URC repository (http://www.timeseriesclassification.com/dataset.php) and noticed some issues with RangerForestClassifier. 

For instance when working with [Ligthning2](http://www.timeseriesclassification.com/description.php?Dataset=Lightning2) dataset , which class labels are [-2,0]
RangerForestClassifier accuracy was very low (e.g., less than 10%) whereas with RandomForestClassifier accuracy was above 70%.

I made a function to transform all class labels to a sequence of positive numbers..for instance.. new class labels for Lightning2 dataset were transformed from [-2 , 0] to [0,  1] and with that change I was able to achieve same classification accuracy as the RandomForestClassifier.

However, with [GunPoint](http://www.timeseriesclassification.com/description.php?Dataset=GunPoint) dataset which class labels are already a sequence of positive number, i.e, [0,1] .. by using skranger classifier the accuracy is around 9% whereas with scikit-learn's random forest it is above 90%..

Not sure how to deal with it...

In case you want to give it a try, you can use the code below and function to read the time series dataset.

```
import pandas as pd
import numpy as np
from scipy.io import arff

dset_name = "Ligthning2"
train_set = "/your/path/to/timeseriesdataset/folder/"+dset_name+"/"+dset_name+"_TRAIN.arff"
test_set = "/your/path/to/timeseriesdataset/folde/"+dset_name+"/"+dset_name+"_TEST.arff"
X_train, y_train, X_test, y_test = getXysets(train_set, test_set)


def getXysets(train_set, test_set):
   
    #get X_train and y_train
    data_train = arff.loadarff(train_set)
    df_train = pd.DataFrame(data_train[0])
    df_train = df_train.dropna() #remove NA

    raw_data = []
    for row in df_train.itertuples():
        raw_data.append(row)
    X_train = []
    y_train = []
    for data in raw_data:
        X_train.append(data[1: len(data)-1])
        y_train.append(int(data[-1])-1)
    X_train = np.array(X_train)
    y_train = np.array(y_train)
    
    #get X_test and y_test
    data_test = arff.loadarff(test_set)
    df_test = pd.DataFrame(data_test[0])
    df_test = df_test.dropna()

    raw_data = []
    for row in df_test.itertuples():
        raw_data.append(row)
    X_test = []
    y_test = []
    for data in raw_data:
        X_test.append(data[1: len(data)-1])
        y_test.append(int(data[-1])-1)
    X_test = np.array(X_test)
    y_test = np.array(y_test)
    
    return X_train, y_train, X_test, y_test
```

**Comment by crflynn** [comment]
No because they should be mapped and unmapped to the natural number sequence [here](https://github.com/crflynn/skranger/blob/cf5ca9da16ecfb1ababa3b7b8d7c315f073fe61e/skranger/ensemble/ranger_forest_classifier.py#L160) and [here](https://github.com/crflynn/skranger/blob/cf5ca9da16ecfb1ababa3b7b8d7c315f073fe61e/skranger/ensemble/ranger_forest_classifier.py#L228), respectively.

If does seem, though, that the returned prediction probabilities from ranger shift the class reference by one, which results in class predictions that are also shifted by one. For example, the following code:

```python
from sklearn.ensemble import RandomForestClassifier
from skranger.ensemble import RangerForestClassifier
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

rf = RandomForestClassifier()
rf.fit(X_train, y_train)

y_pred_rf = rf.predict(X_test)

rfa = accuracy_score(y_test, y_pred_rf)

ra = RangerForestClassifier()
ra.fit(X_train, y_train)

y_pred_ra = ra.predict(X_test)
# shift the prediction values by 1
y_pred_ra = [(v + 1) % 3 for v in y_pred_ra]

raa = accuracy_score(y_test, y_pred_ra)

print(rfa) # 0.98
print(raa) # 0.98
```

I'll see what I can do to fix this.

**Comment by crflynn** [comment]
As a workaround, I think you can modify [this line](https://github.com/crflynn/skranger/blob/cf5ca9da16ecfb1ababa3b7b8d7c315f073fe61e/skranger/ensemble/ranger_forest_classifier.py#L228) in the source code to the following.

```python
        return self.classes_.take((np.argmax(probas, axis=1) + 1) % self.n_classes_, axis=0)
```

Still confused why it's doing this.

**Comment by stevcabello** [comment]
Thanks =)

**Comment by crflynn** [comment]
It looks like ranger also does the exact same mapping, but orders the class_values in order of appearance. I hadn't noticed this but I think the fix is pretty simple and I can take care of it soon.
